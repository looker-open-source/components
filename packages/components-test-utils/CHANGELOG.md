# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
