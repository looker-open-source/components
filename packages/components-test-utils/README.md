# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
