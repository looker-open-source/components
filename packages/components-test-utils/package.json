{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
