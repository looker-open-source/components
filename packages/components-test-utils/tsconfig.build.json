{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
