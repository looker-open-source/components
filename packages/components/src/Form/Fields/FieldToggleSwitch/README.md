# FieldToggleSwitch

The `<FieldToggleSwitch />` component is composed of a `<ToggleSwitch />` component and a `<Label />` component. By default, the label will render to the right of the toggle switch, however this is adjustable with the inline property.

```tsx
function Basic(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    ...restProps
  } = props;

  return <FieldToggleSwitch id={id} label={label} name={name} {...restProps} />;
}
```

## Controlled

```tsx
function Controlled(props: FieldToggleSwitchProps) {
  const {
    on: _on,
    onChange: _onChange,
    label = 'Development Mode',
    ...restProps
  } = props;

  const [on, setOn] = useState(false);
  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
    setOn(e.target.checked);
  };
  return (
    <FieldToggleSwitch
      label={label}
      onChange={onChange}
      on={on}
      {...restProps}
    />
  );
}
```

## Checked

```tsx
function Checked(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    on = true,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      on={on}
      {...restProps}
    />
  );
}
```

## Detail & Description

```tsx
function DetailDescription(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    description = 'describe something here.',
    detail = '4/20',
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      detail={detail}
      description={description}
      {...restProps}
    />
  );
}
```

```tsx
function RichDetailDescription(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    description = (
      <>
        describe something here. <a href="somewhere">Link</a>
      </>
    ),
    detail = <IconButton icon={<MaterialIcons.Delete />} label="Hello world" />,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      detail={detail}
      description={description}
      {...restProps}
    />
  );
}
```

## Disabled

```tsx
function Checked(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    disabled = true,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      disabled={disabled}
      {...restProps}
    />
  );
}
```

```tsx
function Checked(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    disabled = true,
    on = true,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      disabled={disabled}
      on={on}
      {...restProps}
    />
  );
}
```

## Read Only

```tsx
function ReadOnly(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    readOnly = true,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      readOnly={readOnly}
      {...restProps}
    />
  );
}
```

## ValidationMessage

```tsx
function ValidationMessage(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    validationMessage = { message: 'This is an error', type: 'error' },
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      validationMessage={validationMessage}
      {...restProps}
    />
  );
}
```
