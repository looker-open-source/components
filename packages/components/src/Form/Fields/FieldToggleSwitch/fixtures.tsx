/**
 * Copyright (c) 2023 Google LLC
 * SPDX-License-Identifier: MIT
 */

// eslint-disable-next-line no-restricted-imports
import * as MaterialIcons from '@styled-icons/material';
import React from 'react';
import { IconButton } from '../../../Button';
import type { FieldToggleSwitchProps } from '..';
import { FieldToggleSwitch } from '..';

export function RichDetailDescription(props: FieldToggleSwitchProps) {
  const {
    label = 'Toggle Switch',
    name = 'thumbsUp',
    id = 'id',
    description = (
      <>
        describe something here. <a href="somewhere">Link</a>
      </>
    ),
    detail = <IconButton icon={<MaterialIcons.Delete />} label="Hello world" />,
    ...restProps
  } = props;

  return (
    <FieldToggleSwitch
      id={id}
      label={label}
      name={name}
      detail={detail}
      description={description}
      {...restProps}
    />
  );
}
