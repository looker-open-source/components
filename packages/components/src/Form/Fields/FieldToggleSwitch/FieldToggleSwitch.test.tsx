/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import 'jest-styled-components'
import React from 'react'
import { assertSnapshot, mountWithTheme } from '@looker/components-test-utils'
import { FieldToggleSwitch } from './FieldToggleSwitch'

test('A FieldToggleSwitch', () => {
  assertSnapshot(<FieldToggleSwitch id="FieldToggleSwitchID" label="👍" />)
})

test('A FieldToggleSwitch turned on', () => {
  assertSnapshot(
    <FieldToggleSwitch id="FieldToggleSwitchID" label="👍" on={true} />
  )
})

test('A FieldToggleSwitch with validationMessage', () => {
  const wrapper = mountWithTheme(
    <FieldToggleSwitch
      id="FieldToggleSwitchID"
      label="Toggle Switch"
      validationMessage={{ message: 'validation Message', type: 'error' }}
    />
  )
  expect(wrapper.text()).toMatch(`Toggle Switchvalidation Message`)
})

test('A FieldToggleSwitch disabled', () => {
  const wrapper = mountWithTheme(
    <FieldToggleSwitch
      disabled
      id="FieldToggleSwitchID"
      label="Toggle Switch"
    />
  )
  wrapper.find('input').html().includes('disabled=""')
})

test('A FieldToggleSwitch required', () => {
  const wrapper = mountWithTheme(
    <FieldToggleSwitch
      id="FieldToggleSwitchID"
      label="Toggle Switch"
      required
    />
  )
  expect(wrapper.text()).toMatch(`Toggle Switch *`)
})
