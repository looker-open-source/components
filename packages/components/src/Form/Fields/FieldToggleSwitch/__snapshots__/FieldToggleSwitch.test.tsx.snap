// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A FieldToggleSwitch 1`] = `
.c7 {
  -webkit-transition: 300ms;
  transition: 300ms;
  position: absolute;
  bottom: 0.125rem;
  left: 0.125rem;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: #FFFFFF;
}

.c6 {
  -webkit-transition: 300ms;
  transition: 300ms;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 1.25rem;
  background: #C1C6CC;
}

.c6:hover {
  box-shadow: 0 0 .01rem 0.01rem rgba(79,42,186,0.5);
}

.c5 {
  opacity: 0;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.c4 {
  width: 2.1875rem;
  height: 1.25rem;
  display: inline-block;
  position: relative;
  vertical-align: middle;
}

.c4 input {
  cursor: pointer;
}

.c4 input:focus + div {
  box-shadow: 0 0 0 0.2rem rgba(79,42,186,0.4);
}

.c2 {
  color: #343C42;
  font-weight: 600;
  font-size: 0.75rem;
  display: inline-block;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-template-areas: 'input label' '. messages';
  grid-template-columns: repeat(3,max-content);
}

.c0 .c3 {
  grid-area: input;
}

.c0 .c1 {
  font-size: 0.875rem;
  font-weight: normal;
  grid-area: label;
  padding-left: 0.5rem;
}

.c0 .c8 {
  grid-area: messages;
  padding-left: 0.75rem;
}

<label
  className="c0 "
>
  <span
    className="c1 c2"
    color="palette.charcoal700"
    fontSize="xsmall"
    fontWeight="semiBold"
  >
    üëç
  </span>
  <div
    className="c3 "
  >
    <div
      className="c4 "
    >
      <input
        className="c5"
        id="FieldToggleSwitchID"
        role="switch"
        type="checkbox"
      />
      <div
        className="c6"
      >
        <div
          className="c7"
        />
      </div>
    </div>
  </div>
  <div
    className="c8 "
  />
</label>
`;

exports[`A FieldToggleSwitch turned on 1`] = `
.c7 {
  -webkit-transform: translateX(0.9375rem);
  -ms-transform: translateX(0.9375rem);
  transform: translateX(0.9375rem);
  -webkit-transition: 300ms;
  transition: 300ms;
  position: absolute;
  bottom: 0.125rem;
  left: 0.125rem;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: #FFFFFF;
}

.c6 {
  -webkit-transition: 300ms;
  transition: 300ms;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 1.25rem;
  background: #6C43E0;
}

.c6:hover {
  box-shadow: 0 0 .01rem 0.01rem rgba(79,42,186,0.5);
}

.c5 {
  opacity: 0;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

.c4 {
  width: 2.1875rem;
  height: 1.25rem;
  display: inline-block;
  position: relative;
  vertical-align: middle;
}

.c4 input {
  cursor: pointer;
}

.c4 input:focus + div {
  box-shadow: 0 0 0 0.2rem rgba(79,42,186,0.4);
}

.c2 {
  color: #343C42;
  font-weight: 600;
  font-size: 0.75rem;
  display: inline-block;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-template-areas: 'input label' '. messages';
  grid-template-columns: repeat(3,max-content);
}

.c0 .c3 {
  grid-area: input;
}

.c0 .c1 {
  font-size: 0.875rem;
  font-weight: normal;
  grid-area: label;
  padding-left: 0.5rem;
}

.c0 .c8 {
  grid-area: messages;
  padding-left: 0.75rem;
}

<label
  className="c0 "
>
  <span
    className="c1 c2"
    color="palette.charcoal700"
    fontSize="xsmall"
    fontWeight="semiBold"
  >
    üëç
  </span>
  <div
    className="c3 "
  >
    <div
      className="c4 "
    >
      <input
        aria-checked={true}
        checked={true}
        className="c5"
        id="FieldToggleSwitchID"
        role="switch"
        type="checkbox"
      />
      <div
        className="c6"
      >
        <div
          className="c7"
        />
      </div>
    </div>
  </div>
  <div
    className="c8 "
  />
</label>
`;
